src/platform_util.cpp

```diff
@@ -112,7 +112,7 @@ namespace libtorrent {
 		MEMORYSTATUSEX ms;
 		ms.dwLength = sizeof(MEMORYSTATUSEX);
 		if (GlobalMemoryStatusEx(&ms))
-			ret = int(ms.ullTotalPhys);
+			ret = ms.ullTotalPhys;
```

Comment:
Technically, isn't there still a possibility of unsigned -> signed conversion which would return a negative number? If the number is bigger than `std::numeric_limits<int64_t>::max()`, just truncate to that value, so that a negative number is never returned.

Reply:
you're right. I assumed `DWORDLONG` was signed

Reply:
interestingly, there is no warning for this. I think it's safe to assume the amount of memory will not exceed the max for `int64_t` (given that current hardware only can address 48 bits)

Reply:
Still I think explicitly enforcing the the truncation to `std::numeric_limits<int64_t>::max()` better conveys the intent for humans, even if the machines don't need it.